core/app/models/spree/line_item.rb

```diff
@@ -106,16 +106,19 @@ def insufficient_stock?
       !sufficient_stock?
     end
 
-    # Sets the options on the line item according to the order's currency or
-    # one passed in.
+    # Sets options on the line item.
+    #
+    # The option can be arbitrary attributes on the LineItem. If no price is given in the options,
+    # this will call the legacy `PriceModifier` line item pricer.
     #
     # @param options [Hash] options for this line item
     def options=(options = {})
       return unless options.present?
 
-      self.price = variant.price_in(currency).amount +
-                   variant.price_modifier_amount_in(currency, options)
-
+      # There's no need to call a pricer if we'll set the price directly.
+      unless options[:price].is_a?(Numeric)
```

Comment:
Is it common to have a price passed in here?

I'd prefer checking just `options.has_key?(:price)`, as both `line_item.price = '12.34'` and `line_item.price = nil` do work. Even if we don't expect options to be used that way, it's harder to reason about if it behaves differently based on the type of `:price` provided